lib/core/ui/dashboard/home/home_calendar.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:live_score/core/constants/dashboard_constants.dart';
+import 'package:live_score/core/theme/app_theme.dart';
+import 'package:live_score/core/ui/dashboard/home/home_date_element.dart';
+import 'package:live_score/core/utils/date_util.dart';
+import 'package:live_score/features/dashboard/cubit/dashboard_cubit.dart';
+
+class DashboardCalendar extends StatelessWidget {
+  const DashboardCalendar({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    List<DateTime> weekDates =
+        DateUtil.generateDates(DashboardConstants.calendarRange);
+    final ScrollController scrollController = ScrollController();
+    final cubit = context.read<DashboardCubit>();
+
+    WidgetsBinding.instance.addPostFrameCallback(
+      (_) {
+        final DateTime now = DateTime.now();
+        final int todayIndex = weekDates.indexWhere((date) =>
+            date.year == now.year &&
+            date.month == now.month &&
+            date.day == now.day);
+
+        if (todayIndex != -1) {
+          const double itemWidth = DashboardConstants.calendarItemWidth;
+          final double screenWidth = MediaQuery.of(context).size.width;
+          final double offset =
+              (todayIndex * itemWidth) - (screenWidth / 2) + (itemWidth / 2);
+
+          scrollController.jumpTo(offset);
+
+          cubit.setScrollOffset(offset);
+        }
+      },
+    );
+
+    void scrollLeft() {
+      scrollController.animateTo(
+        scrollController.offset - DashboardConstants.calendarScrollOffset,
+        duration: const Duration(
+          milliseconds: DashboardConstants.calendarScrollDuration,
+        ),
+        curve: Curves.easeInOut,
+      );
+    }
+
+    void scrollRight() {
```

Comment:
These functions `scrollRight` `scrollLeft` etc shouldn't be placed inside the `build` method. Please move them.

Reply:
Simply move out of build context is enough or should I create an util file for that ?

Reply:
i moved it to DateUtil file that i created previously, it does fit there i believe